### PR TITLE
API Generator: Disable input generation if `create` and `update` are set to `false` in the decorator

### PR DESCRIPTION
**Pull request by VP-DS**
https://github.com/vivid-planet/comet/assets/118435139/c38eece2-afcd-487c-80f3-ffe02e7eed87

COM-419

### CODE REVIEW

File: packages/api/cms-api/src/generator/crud-generator.decorator.ts [hunk 1]
```diff
@@ -22,6 +22,9 @@ export function CrudGenerator({
 export interface CrudSingleGeneratorOptions {
     targetDirectory: string;
     requiredPermission?: string[] | string;
+    create?: boolean;
+    update?: boolean;
+    delete?: boolean;
```
Comment (nsams) [inline packages/api/cms-api/src/generator/crud-generator.decorator.ts:27]: Single Generator doesn't generate create or delete, please remove that.

Comment (VP-DS) [reply packages/api/cms-api/src/generator/crud-generator.decorator.ts:27]: Ah, I see it does not affect the resolver.

**Review by nsams — APPROVED** [review]

File: packages/api/cms-api/src/generator/generate-crud-single.ts [hunk 2]
```diff
@@ -7,6 +7,10 @@ import { GeneratedFile } from "./utils/write-generated-files";
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export async function generateCrudSingle(generatorOptions: CrudSingleGeneratorOptions, metadata: EntityMetadata<any>): Promise<GeneratedFile[]> {
+    generatorOptions.update = generatorOptions.update ?? true;
```
[the remark this answers is not in the document]
Comment (VP-DS) [reply packages/api/cms-api/src/generator/generate-crud-single.ts]: https://github.com/vivid-planet/comet/pull/1675

File: .changeset/atlantic-scallop-fisherman [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-api": patch
+---
+
+API Generator: Disable input generation if `create` and `update` are set to false in the crud decorator
```
Comment (johnnyomair) [inline .changeset/atlantic-scallop-fisherman:5]:
```suggestion
API Generator: Disable input generation if `create` and `update` are set to `false` in the decorator
```

**Review by nsams — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**